### ISSUE TITLE
Unable to import MBOXes

### ISSUE THREAD

**Issue by Jo-suis-foutu** [thundernest/import-export-tools-ng#15]
Thank you for updating this essential tool! I am still unable to import MBOX files though. That is, the folder shows up but there are no emails in it. This is on Linux Mint 18.3 with Thunderbird 60.8.0. Very frustrating! thank you.

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
Thanks for posting the issue.  What version of the extension are you using?  I just did an update that fixes that issue with EML that also affected other imports.
I literally just posted to the review queue.
If you would like to test it, download the XPI 4.0.3 from the xpi repository folder
Feedback very much appreciated!

**Comment by Jo-suis-foutu** [comment]
OK, thanks. I downloaded and installed 4.0.3 but it didn't fix the problem. Is there a difference between the three different XPIs numbered 4.0.3?

**Comment by Jo-suis-foutu** [comment]
Hmm. Maybe this is a problem with the MBOX files. I did import a different one. They kind of look OK and TBird doesn't announce they're corrupt or anything, and yet no emails show up in the folders.

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
https://github.com/thundernest/import-export-tools-ng/blob/master/xpi/import-export-tools-ng-4.0.3-tb.xpi
I just retried my Windows box and the Mbox files imported okay.
I don't currently have Linux setup, maybe the OS is the issue?
did you try doing an export then importing the same file?

**Comment by Jo-suis-foutu** [comment]
I don't have anything to export from. these are supplied to me by parties to legal matters. I wonder if you could try this one, which doesn't work for me, but seems OK otherwise. If OS is the issue I can deal with it:

https://www.dropbox.com/s/hp450ku5hx7xaq6/CPRA_request_%28SPPS.2019.08.12.b%29_jwatts%40scholarshipschools.org_0.mbox?dl=0

**Comment by cleidigh** [comment]
I will try that right now.
appreciate the help on this..

**Comment by cleidigh** [comment]
Nope
it's not the  OS, fails for me also.
it's quite a large file. I will do some further testing.
you could also try exporting using the extension itself and then trying to import.

**Comment by Jo-suis-foutu** [comment]
OK, thanks. I will complain to my source. I routinely import multi-GB mboxes, so this one doesn't seem large. Not sure what you mean about exporting using the extension...

**Comment by cleidigh** [comment]
you can export to a mbox file using the extension option:
Export folder to mbox.
it was slightly unclear above , have you been successful importing other files with 4.0.3?
I will also continue testing, thanks.

**Comment by Jo-suis-foutu** [comment]
I tried that, although it's called "Export folder" in my menu. It said it did it but it didn't save a file. Also the properties of the folder show 0 bytes, so there seems to be nothing in there. 

Also, yes, I did just now manage to import a 2.2 GB MBOX using 4.0.3.  It's not unheard of for my sources to mess up exports, sometimes on purpose, so I will get them to redo these. 

Thank you so much for your help!

**Comment by cleidigh** [comment]
I also just tried the original extension v3.3.2 and reports a problem as well.
I am still concerned about your failure to export the folder.  that works fine under Windows for me
so I would like to make sure to address that if it's truly a problem.  do you see any errors in the console?

**Comment by Jo-suis-foutu** [comment]
It's the filenames! I renamed them all to <single digit>.mbox and they imported fine! Probably all the weird symbols in there are messing something up!

**Comment by Jo-suis-foutu** [comment]
Sorry, renamed them all to like 0.mbox, 1.mbox, and so on, and then they imported

**Comment by cleidigh** [comment]
which "files"
inside the main file?

**Comment by Jo-suis-foutu** [comment]
No, sorry. I have 6 of these mboxes. I just sent you the smallest one. But try renaming it to like 0.mbox and see what happens

**Comment by cleidigh** [comment]
Great, if I change the name I import okay also.
I will do  debugging tomorrow to see what characters are throwing things off.
if it's not too much trouble at some point to look at the export issue if you get any errors.
This extension is key for Thunderbird which is why I took it over as part of the Thunderbird team.
Because so many things are changing it is easy for things to break unfortunately.

**Comment by Jo-suis-foutu** [comment]
Export worked perfectly! I know this is a bug, but I am so glad to have a workaround. I can't thank you enough for your help and your support of this crucial extension.

**Comment by cleidigh** [comment]
Just to be clear, Folder Export (mbox) execute successfully.
We agree with your original  issue  is apparently a file naming issue in the extension, correct?
Very happy it's a minor issue with the workaround.  I am sure I can fix it without too much trouble for the next maintenance Release.

**Comment by Jo-suis-foutu** [comment]
Yes, that's correct. I renamed one of the problematic ones to 0.mbox and it imported fine. Then I was able to use Folder Export to re-export an MBOX.

**Comment by cleidigh** [comment]
excellent.
I appreciate going back and forth with you.
Once I have a beta I will note it here so you can try it before I publish it.
Thanks again!

**Comment by Jo-suis-foutu** [comment]
Thank you so much for your help on this!

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
No problem!  look for 4.0.4 shortly.
If you could do review that would be great!
@cleidigh

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
I think I can go ahead and close this yes?
Could you do a review?
Thanks

**Comment by Jo-suis-foutu** [comment]
Yes, and yes. Thank you so much for your help!

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
ThankYou!

**Comment by Jo-suis-foutu** [comment]
Hey, how are you?

and it seems to have happened again with importing mboxes. Here is one that wouldn't import, although the names of the emails were visible:

https://send.firefox.com/download/7864f9cf4f144f60/#d9C48bnvQNMx43q2Btyd1A

Also a screenshot of the error message. But when I renamed it to 0.mbox everything worked fine.

Thanks again for all your help. Is it possible to donate money to you or elsewhere for this excellent work?

Adrian (my real name)

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
Arrg! Sorry
is this still with Thunderbird 68.1 or are you on BetaSomething?  and I assume 4.0.4?
BTW I just saw this but I am getting offline but I want to look at it tomorrow

**Comment by Jo-suis-foutu** [comment]
Sorry for omitting the details. This is 60.9.0, which is what my distribution seems to think I ought to have. Maybe that's the problem? I'm not sure what you mean by 4.0.4? Take your time, of course. Thanks.

**Comment by cleidigh** [comment]
well I omitted details as well...
I meant the version of the extension.
so nothing changed as far as what you are using and what we worked on before?

**Comment by Jo-suis-foutu** [comment]
Ah, I see. Yes, 4.0.4. And it's true, nothing changed from what we worked on before.

**Comment by cleidigh** [comment]
okay I will work on that

**Comment by cleidigh** [comment]
Adrian
Thanks for the proposition.  I am a retired disabled engineer who has volunteered to serve as Thunderbirds extension developer liaison, in addition to adopting various extensions myself.
A donation would be very kind but uneeded.  I am sure Thunderbird can use the donations if that suits you.

I just tested your file from dropbox using Thunderbird 60.9/IETNG 4.0.4 without changing anything especially the filename , imported just fine.  You mentioned screenshot of an error, can you post that?
I thought it might be an issue with 4.0.5-b1 which is posted here but not released yet.
obviously we will have to dig deeper.  First I have to be able to reproduce...

**Comment by Jo-suis-foutu** [comment]
Yes, sure, here's the screenshot. 

/home/ginsberg/Downloads/Screenshot from 2019-10-10 17-59-52.png

**Comment by cleidigh** [comment]
@Jo-suis-foutu  
I'm not able to reproduce the error.  Does this error come up each time you click on a message?
This may be a platform issue.  I think I need to spin up a virtual machine.
on your side can you try importing to a top level folder with a short name, initially keeping box file same.  I'm just looking at the combined length is very long, maybe some other limit somehow different on Windows versus. mint

**Comment by cleidigh** [comment]
@Jo-suis-foutu 
Any more useful information on this?

**Comment by Jo-suis-foutu** [comment]
Hi, sorry. It slipped off my radar. I will respond soon.
